File: markdowner-core/src/main/java/org/jesperancinha/parser/markdowner/MergeParserHelper.java
package org.jesperancinha.parser.markdowner;

import org.jesperancinha.parser.markdowner.model.Paragraphs;
import org.jesperancinha.parser.markdowner.model.Paragraph;

import java.util.List;
import java.util.stream.Collectors;

import static java.lang.System.lineSeparator;

public class MergeParserHelper {

    /**
     * Receives a complete markdown text and a {@link Paragraphs} instance and adds all paragraphs in the stipulated order to the end of the text
     *
     * @param readmeMd A complete String representation of a markdown text
     * @param footer   A {@link Paragraphs} instance which will add all paragraphs to the end of the markdown text
     * @return The complete String merge between a {@link String} text and a {@link Paragraphs} instances
     */
    public static String mergeDocumentWithFooterTemplate(String readmeMd, Paragraphs footer) {
        List<String> paragraphsBuildList = footer.getTags()
                .stream()
                .map(footer::getParagraphByTag)
                .map(Paragraph::toString)
                .collect(Collectors.toList());
        return readmeMd.concat(lineSeparator()).concat(lineSeparator()).concat(String.join(lineSeparator(), paragraphsBuildList));
    }
}

File: markdowner-core/src/main/kotlin/org/jesperancinha/parser/markdowner/helper/MergeParserHelper.kt
package org.jesperancinha.parser.markdowner.helper

import org.jesperancinha.parser.markdowner.model.Paragraphs

object MergeParserHelper {
    /**
     * Receives a complete Markdown text and a [Paragraphs] instance and adds all paragraphs in the stipulated order to the end of the text
     *
     * @param readmeMd A complete String representation of a Markdown text
     * @param footer   A [Paragraphs] instance which will add all paragraphs to the end of the Markdown text
     * @return The complete String merge between a [String] text and a [Paragraphs] instances
     */
    fun mergeDocumentWithFooterTemplate(readmeMd: String, footer: Paragraphs) =
        readmeMd + System.lineSeparator() + System.lineSeparator() + java.lang.String.join(
            System.lineSeparator(),
            footer.tags
                .map { tag: String -> footer.getParagraphByTag(tag).toString() }
        )
}